### PR TITLE
Add implementation of production and consumption stats RPCs

### LINKED ISSUE #82: Add RPC call for querying statistics for nodes from Node Executor
I just noticed that we're still missing an RPC call from the executor to the instance to query for production/consumption stats for each node. We should address that (probably in a followup PR) as it is required for the scaling algorithm.

_Originally posted by @pulsastrix in https://github.com/hsel-netsys/iceflow/issues/79#issuecomment-2286377709_

### PR DESCRIPTION
**Pull request by JKRhb**
Still WIP.

Depends on #86 and #87.

Will eventually resolve #82.

### CODE REVIEW

**Comment by JKRhb** [comment]
@pulsastrix I think this PR should now be a ready for an initial review. As discussed yesterday, it is now possible to associate one producer and multiple consumers with the NodeInstance service, the consumers can be differentiated via the `edgeName`. The way the production stats are collected probably needs some more discussion, looking forward to your opinion here.

To achieve a cleaner diff and focus on the actual changes associated with this PR, #86 and #87 should get merged first before conducting an actual review, though.

**Comment by pulsastrix** [comment]
@JKRhb I will take a look at it as soon as #87 is merged.

> As discussed yesterday, it is now possible to associate one producer and multiple consumers with the NodeInstance service, the consumers can be differentiated via the edgeName. 

A single node instance might have multiple outgoing edges with differing data (e.g. one output for video frames, one for audio frames, ...), in which case we might need more than one producer for a single instance.

**Comment by JKRhb** [comment]
> A single node instance might have multiple outgoing edges with differing data (e.g. one output for video frames, one for audio frames, ...), in which case we might need more than one producer for a single instance.

Hmm, yeah, good point, I will revert this aspect of the PR.

File: src/consumer.cpp [hunk 1]
```diff
@@ -35,7 +37,36 @@ IceflowConsumer::IceflowConsumer(std::shared_ptr<IceFlow> iceflow,
 IceflowConsumer::~IceflowConsumer() { unsubscribeFromAllPartitions(); }
 
 std::vector<uint8_t> IceflowConsumer::receiveData() {
-  return m_inputQueue.waitAndPopValue();
+  auto value = m_inputQueue.waitAndPopValue();
+
+  // TODO: Discuss if this should be put elsewhere
+  saveTimestamp();
```
Comment (pulsastrix) [inline src/consumer.cpp:43]: Should be fine to keep it here.

```suggestion
  saveTimestamp();
```

**Review by pulsastrix — APPROVED** [review]

**Pull request merged**